Fix linebreaks

ixs opened the pull request:
Soft Line Breaks are not handled optimally in Mail_Mime.

* Breaks happen too early at char 76 rather than _after_ char 76 meaning that a line gets broken up too early.
* In the situation that a dot would end up as the first char on a new line extra handling and extra escaping is needed. This calculation suffers from another miscalculation reducing the line length by one more char.

Fix both these issues.
Add tests for to both issues.

ixs commented:
Mhm. Just found another weird edge case that fails. Need to have another look and figure it out...

ixs commented:
Alright. Better now. 😁

till commented on tests/encoding_case.phpt:
```diff
@@ -8,6 +8,5 @@ require_once('Mail/mimePart.php');
 --EXPECT--
-Here's=
-=09
+Here's=09
```
@ixs Email is not really my problem domain anymore. I see you adjusted another test, is that save to do?

@pear/mail-components Anyone else wants to weigh in? LGTM otherwise.

ixs replied:
@till 😁 That test broke after my change.
I had a hard and long look at it, fixed a bug in my code and then looked again at the test.

I was just explaining what exactly the old code did and why it did it wrong here... But while writing that down, I'm having a few doubts about and think I might have missed an interesting case... So let me have another hard look at this test and get back to you in a few. 😝

till commented on tests/test_linebreak_dot.phpt:
```diff
@@ -0,0 +1,48 @@
+--TEST--
```
🎉 Yay, new test!

till commented on tests/test_linebreak_larger_76.phpt:
```diff
@@ -0,0 +1,59 @@
+--TEST--
```
👍

ixs commented:
Pushed a new version.

@till So I had another hard look at the code and I missed one case indeed. I did not read the tab test results correctly initially. So thanks for asking and making me have another look. 👍🏻

Reverted the change to the tests/encoding_case.phpt file and added one more test file that I used during debugging. It's partly duplicating existing tests but it contains a nice print function that is good for human consumption...

till reviewed (approved):
This LGTM.

Maybe @alecpl wants to take a look?

alecpl commented on tests/qp_encoding_test.phpt:
```diff
@@ -0,0 +1,98 @@
+        for($j=0; $j < count($lines); $j++) {
+            $line = $lines[$j];
+            if ($j + 1 < count($lines) && $special_chars) {
+                $line_vis = str_replace("\t", '\t', $line).'\r\n';
```
Would be nice to have this str_replace() used for printing the input too.

ixs replied:
It makes it easier to read when tabs are involved, but with newlines it's now one long string...

But ehh, it's six of one, half a dozen of the other... Added the change.

alecpl commented on tests/qp_encoding_test.phpt:
```diff
@@ -0,0 +1,98 @@
+---
+input: 15: 	Here's	
+a tab.
+output:08: 	Here's=
```
I would expect to see `\t` here in the start of output.
